db/background_mgr.go

```diff
@@ -124,23 +298,32 @@ func (b *BackgroundManager) Stop() error {
 		return err
 	}
 
-	b.terminator <- struct{}{}
+	b.Terminate()
 	return nil
 }
 
+// Terminate stops the process via terminator channel
+// Only to be used internally to this file and by tests.
+func (b *BackgroundManager) Terminate() {
+	if !b.terminatorClosed {
```

Comment:
Since there's no synchronization here, this doesn't look like it's going to prevent the channel being closed twice (if that's the intention).

Can we avoid the extra terminatorClosed by setting the status (to stopping) here w/ a compareAndSwap, and only closing the terminator if the state isn't already stopping?